core/devdoc/message_queue_requirements.md

```diff
@@ -0,0 +1,128 @@
+MESSAGE QUEUE REQUIREMENTS
+==========================
+
+Overview
+--------
+
+The message queue is a very simple queue intended to manage messages. The message queue is typed with MESSAGE_HANDLE because the destruction of the queue requires the destruction of the messages inside the queue.  **Unless the queue is destroyed, the user of this queue is expected to clone before pushing onto the queue, and is expected to destroy the message after popping the message off teh queue.**
```

Comment:
>The message queue is a very simple queue intended to manage messages. The message queue is typed with MESSAGE [](start = 0, length = 109)

Sorry for this comment. But do we really need this? Why don't we use one of the existing lists that we have already on CShared Utility?